tasks/service-runner/app/workers/pull-worker.ts

```diff
@@ -142,29 +142,32 @@ async function _pullAndDoWork(repeat = true): Promise<void> {
     logger.error(e);
   }
 
-  // check to see if we are terminating
-  const terminationFilePath = path.join(LOCKFILE_DIR, 'TERMINATING');
   try {
-    await fs.access(terminationFilePath);
-    // TERMINATING file exists so PreStop handler is requesting termination
-    logger.debug('RECEIVED TERMINATION REQUEST');
-    // removing the WORKING file is done in the `finally` block at the end of this function
-    return;
-  } catch {
-    // expected if file does not exist
-  }
+    // check to see if we are terminating
```

Comment:
Had to move this inside this try block to make sure the WORKING file gets cleaned up.